src/resources/js/admin/help-hub-iframe.js

```diff
@@ -76,125 +81,189 @@ window.DocsBotAI = window.DocsBotAI || {};
-		const isOptedIn = bodyElement.getAttribute( 'data-opted-in' ) === '1';
-		const optOutMessageElement = document.querySelector( obj.selectors.optOutMessage );
-		const docsbotElement = document.getElementById( obj.selectors.docsbotWidget );
-		// Only run Zendesk and DocsBot setup if the user has opted-in.
-		if ( isOptedIn ) {
-			obj.loadAndInitializeZendeskWidget();
-			obj.initializeDocsBot();
-		} else {
-			optOutMessageElement.classList.remove( 'hide' );
-			docsbotElement.classList.add( 'hide' );
-			bodyElement.classList.add( 'blackout' );
-		}
+	obj.getHelpHubPageElement = function() {
+		return document.getElementById(obj.selectors.helpHubPageID);
 	};
 
 	/**
-	 * Dynamically loads the Zendesk Web Widget script.
+	 * Utility: Check if user is opted in.
 	 *
-	 * @since 6.3.2
-	 * @param {string} zendeskKey - The Zendesk chat key.
-	 * @return {Promise} - A promise that resolves when the script is loaded.
+	 * @since TBD
+	 * @return {boolean}
 	 */
-	obj.loadZendeskWidgetScript = ( zendeskKey ) => {
-		return new Promise( ( resolve, reject ) => {
-			const script = document.createElement( 'script' );
-			script.id = 'ze-snippet';
-			script.src = `https://static.zdassets.com/ekr/snippet.js?key=${ zendeskKey }`;
-			script.async = true;
-
-			document.head.appendChild( script );
-
-			script.onload = () => resolve();
-			script.onerror = () => reject( new Error( 'Failed to load Zendesk Web Widget' ) );
-		} );
+	obj.isOptedIn = function() {
+		var el = obj.getHelpHubPageElement();
+		return el && el.getAttribute('data-opted-in') === '1';
 	};
 
 	/**
-	 * Initializes the Zendesk widget, hides it initially, and sets up event listeners for open/close actions.
+	 * HelpScoutManager handles Help Scout Beacon integration.
 	 *
-	 * @link https://support.zendesk.com/hc/en-us/articles/4408836216218-Using-Web-Widget-Classic-to-embed-customer-service-in-your-website
+	 * @since TBD
+	 * @class
+	 * @link https://developer.helpscout.com/beacon-2/web/javascript-api
+	 */
+	function HelpScoutManager(beaconKey, userIdentifiers) {
+		this.beaconKey = beaconKey;
+		this.userIdentifiers = userIdentifiers || null;
+		this.scriptLoaded = false;
+		this.beaconReady = false;
+		this.initPromise = null;
+	}
+
+	/**
+	 * Loads the Help Scout Beacon script dynamically.
 	 *
-	 * @since 6.3.2
-	 * @return {void}
+	 * @since TBD
+	 * @return {Promise<void>}
 	 */
-	obj.initializeZendesk = () => {
-		obj.isZendeskInitialized = false;
-		const bodyElement = document.getElementById( obj.selectors.helpHubPageID );
+	HelpScoutManager.prototype.loadScript = function() {
+		const self = this;
+		if (self.scriptLoaded) return Promise.resolve();
 
-		zE( 'webWidget', 'hide', () => {
-			obj.isZendeskInitialized = true;
-		} );
+		// Replicate the official Help Scout Beacon stub logic (ES6)
+		if (!window.Beacon || !window.Beacon.readyQueue) {
+			window.Beacon = function(method, options, data) {
```

Comment:
Is this a name that we fully control? If so, `window.Beacon` seems a bit generic to me. It would be nice to see some kind of namespacing.